### PR TITLE
added functionality for google sign up button

### PR DESCRIPTION
**Pull request by rajanibusani**
# Description
In the RegistrationPage, user can register by clicking Google signup button, 
Please provide a short summary explaining what this PR is about.

Fixes # (issue)

# How to test?
npm run dev

Please provide a short summary how your changes can be tested?

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/GoogleLoginButton/GoogleLoginButton.component.js [hunk 1]
```diff
@@ -1,14 +1,47 @@
...
+  const history = useHistory();
+
+  const { auth, signInGoogle, googleSignOut } = useFirebase();
+
+  const onClick = async () => {
+    setIsLoading(true);
+    const name = await signInGoogle();
+    setUserName(name);
+    setIsLoading(false);
+    console.log(auth);
```
Comment (bytewiz) [inline src/client/components/GoogleLoginButton/GoogleLoginButton.component.js:22]: We don't put logs when submitting PRs :) 🧹

File: src/client/components/GoogleLoginButton/GoogleLoginButton.component.js [hunk 2]
```diff
@@ -1,14 +1,47 @@
...
+    // alert(name)
+  };
+  const google = () => {
+    setUserName('');
+    const abc = googleSignOut();
+    history.push('/');
+    console.log('sign out button', abc);
```
Comment (bytewiz) [inline src/client/components/GoogleLoginButton/GoogleLoginButton.component.js:29]: We don't put logs when submitting PRs :) 🧹

File: src/client/components/GoogleLoginButton/GoogleLoginButton.component.js [hunk 3]
```diff
@@ -1,14 +1,47 @@
...
+    setIsLoading(true);
+    const name = await signInGoogle();
+    setUserName(name);
+    setIsLoading(false);
+    console.log(auth);
+    // alert(name)
+  };
+  const google = () => {
+    setUserName('');
+    const abc = googleSignOut();
```
Comment (bytewiz) [inline src/client/components/GoogleLoginButton/GoogleLoginButton.component.js:27]: What is `abc` not the best name doesn't explain me anything :)

File: src/client/components/GoogleLoginButton/GoogleLoginButton.component.js [hunk 4]
```diff
@@ -1,14 +1,47 @@
...
 const GoogleLoginButton = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [userName, setUserName] = useState('');
+  const history = useHistory();
+
+  const { auth, signInGoogle, googleSignOut } = useFirebase();
+
+  const onClick = async () => {
+    setIsLoading(true);
+    const name = await signInGoogle();
```
Comment (bytewiz) [inline src/client/components/GoogleLoginButton/GoogleLoginButton.component.js:19]: So sign in returns the name?

File: src/client/firebase/FirebaseContext.js [hunk 5]
```diff
@@ -36,13 +37,16 @@ export function FirebaseProvider({ children, initialAuth }) {
...
       isInitialized: !!auth,
       signIn: (data) => signIn(auth, data),
       signUp: (data) => signUp(auth, data),
       signOut: () => signOut(auth),
       resetPassword: (data) => resetPassword(auth, data),
+      signInGoogle: () => signInGoogle(),
+      googleSignOut: () => googleSignOut(),
     }),
-    [auth],
+    [],
```
Comment (bytewiz) [inline src/client/firebase/FirebaseContext.js:49]: Why are you not including the `auth` object in the hook dependencies?

File: src/client/firebase/GoogleAuth.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  googleProvider: new firebase.auth.GoogleAuthProvider(),
+};
+
+export async function signInGoogle() {
+  // const { setAuth } = useFirebase();
+  try {
+    const result = await firebase
+      .auth()
+      .signInWithPopup(providers.googleProvider);
+    await console.log(result.user.displayName, result);
```
Comment (bytewiz) [inline src/client/firebase/GoogleAuth.js:15]: no logs in the PRs :) also, why did you add `await` for the log? 🤔

File: src/client/hooks/useAuthentication.js [hunk 7]
```diff
@@ -32,6 +34,10 @@ export function useAuthentication() {
       if (user) {
         setIsAuthenticated(true);
         setIsLoading(false);
+
+        const uid = user.uid;
```
Comment (bytewiz) [inline src/client/hooks/useAuthentication.js:38]: Could be written like this but it is up to you :) 
```suggestion
        const { uid } = user;
```

File: src/client/hooks/useAuthentication.js [hunk 8]
```diff
@@ -32,6 +34,10 @@ export function useAuthentication() {
...
+        console.log(uid);
```
Comment (bytewiz) [inline src/client/hooks/useAuthentication.js:39]: No logs 🧹

**Review by bytewiz — CHANGES_REQUESTED** [review]

File: src/client/components/GoogleLoginButton/GoogleLoginButton.component.js (hunk 4 above)
Comment (rajanibusani) [reply src/client/components/GoogleLoginButton/GoogleLoginButton.component.js:19]: Yes, In signIn function , i returned user.displayname, and storing in the name

File: src/client/hooks/useAuthentication.js (hunk 7 above)
Comment (rajanibusani) [reply src/client/hooks/useAuthentication.js:38]: i will, Tq niels

File: src/client/hooks/useAuthentication.js (hunk 8 above)
Comment (rajanibusani) [reply src/client/hooks/useAuthentication.js:39]: sorry, i will remove them
